weasyl/spam_filtering.py

```diff
@@ -0,0 +1,139 @@
+import os
+
+from akismet import Akismet, SpamStatus
+
+from weasyl import config
+from weasyl import define as d
+from weasyl.error import WeasylError
+
+
+FILTERING_ENABLED = config.config_read_bool(setting='enabled', value=False, section='spam_filtering')
+AKISMET_KEY = config.config_read_setting(setting='key', value=None, section='spam_filtering')
+# Used to set the `is_spam` flag on requests sent to the backend
+_IS_ENVIRONMENT_TESTING = True if os.environ.get("WEASYL_TESTING_ENV") else False
+# AKA, the link to the main page of the web application
+AKISMET_BLOG_URL = "http://lo.weasyl.com" if _IS_ENVIRONMENT_TESTING else "https://www.weasyl.com"
+
+
+if all([FILTERING_ENABLED, AKISMET_KEY]):
+    _akismet = Akismet(
+        AKISMET_KEY,
+        blog=AKISMET_BLOG_URL,
+        application_user_agent="weasyl/+({0})".format("https://github.com/weasyl/weasyl")
+    )
+else:
+    FILTERING_ENABLED = False
+
+
+def _check_if_filtering_enabled(function):
```

Comment:
This appears to be used only once, and I don’t think it’s clearer than just having an inline

```python
def submit(
    ⋮
):
    if not FILTERING_ENABLED:
        raise WeasylError("SpamFilteringDisabled")

    ⋮
```